DMTN-148.tex

```diff
@@ -50,9 +50,276 @@
+\verb|OBSTYPE| keyword is universally populated in all calibrations,
+regardless of if they are generated by constructCalibs.py or are
+curated by the \verb|obs_CAMERA| packages.  Curated calibrations also
+require the \verb|DETECTOR| and \verb|CALIBDATE|, the last of which is
+also populated by constructCalibs.py.  The \verb|CALIB_ID| is set by
+constructCalibs.py, and exists for the majority of curated
+calibrations, but is not consistently used.
+
+The keywords listed in table \ref{tab:metadata} are the proposed list
+of header keywords that will be required in the new operational
+calibrations in Gen3.  The majority of these are inserted at creation,
+identifying the content and applicable data range for the calibration.
+Only upon certification (the replacement for the Gen2 ingest) are the
+valid date ranges set.  To conform with the expected changes to FITS
+persistence in RFC-686, these and all other calibration-specific
+metadata fields will be upper-case.
+
+\begin{tabular}{l l l}
+  \label{tab:metadata}
+  Keyword & Inserted & Description \\
+  \hline
+  \verb|OBSTYPE| & Creation & Calibration dataset type contained in this file. \\
+  \verb|{OBSTYPE}_SCHEMA| & Creation & Storage schema for calibration.  \\
+  \verb|{OBSTYPE}_VERSION| & Creation & Storage schema version. \\
+  \verb|INSTRUME| & Creation & Instrument/camera this calibration belongs to. \\
+  \verb|DETECTOR| & Creation & Integer identifier for the ccd/detector. \\
+  \verb|FILTER| & Creation & Name of the physical filter or ``NONE''. \\
+  \verb|CALIB_ID| & Creation & DataId that can be used to request this calibration after ingest. \\
+  \verb|CALIB_CREATION_DATE| & Creation & Creation date of the calibration. \\
+  \verb|CALIB_CREATION_TIME| & Creation & Creation time of the calibration. \\
+  \verb|CALIBDATE| & Creation & Concatenation of \verb|CALIB_CREATION_DATE| and \verb|CALIB_CREATION_TIME|. \\
+  \verb|VALIDSTART| & Certification & Start date the calibration is valid from.  \\
+  \verb|VALIDEND| & Certification & End date the calibration is valid from.  \\
+\end{tabular}
+
+
+\subsection{Calibration Construction in Gen3}
+
+Calibration products should be constructed using well defined
+PipelineTask code.  This removes the need to delineate between
+``calibrations'' and ``user-curated calibrations''.  The only
+remaining difference between these two groups is what the initial
+persisted data type is: ``calibrations'' are stored in FITS images, as
+they represent corrections that are only meaningful as full images;
+``user-curated calibrations'' can be visualized in a small number of
+parameters, and so persisting these in a text format to allow user
+inspection is reasonable.  ECSV and YAML are good formats for these
+calibrations, and a calibration's readText/writeText methods should
+implement one of these.
+
+In addition to the calibration product, the PipelineTask code must
+persist a record of the input data used for construction.  This, along
```

Comment:
Isn't this record of input data redundant with Gen3 provenance?  Is this a write-only product, or is it ingested/used?  Which is the source of truth?

Reply:
My goal here was to allow something to be used to export the "recipe" used to make the calibrations for an external butler.  This is almost certainly redundant with the Gen3 provenance, and should probably be defined as an exportable dump from the butler.